File: php/src/Exports/AgentJson.php
<?php
declare(strict_types=1);

namespace Bkbs\Exports;

/**
 * Stage 4c: honest knowledge index. No stub protocol, no A2A endpoint.
 */
final class AgentJson
{
    /**
     * @param array<string, mixed> $site
     * @param list<array<string, mixed>> $entities
     * @return array<string, mixed>
     */
    public static function build(array $site, array $entities): array
    {
        unset($entities);
        $base = rtrim((string) $site['base_url'], '/');
        return [
            'name' => $site['name'],
            'url' => $site['base_url'],
            'knowledge' => [
                'llms_txt' => $base . '/llms.txt',
                'llms_full' => $base . '/llms-full.txt',
                'graph' => $base . '/graph.json',
                'schema_organization' => $base . '/schema/organization.jsonld',
                'schema_services' => $base . '/schema/services.jsonld',
            ],
        ];
    }
}

File: app/exports/agent_json.py
"""/.well-known/agent.json — honest knowledge index (Stage 4c).

Not an A2A agent card: no stub protocol, no endpoint, no fake capabilities list.
"""

from __future__ import annotations

from typing import Any, Sequence


def build_agent_json(site: Any, entities: Sequence[Any]) -> dict[str, Any]:
    """Knowledge-index pointers for origin files. `entities` unused (signature stable)."""
    _ = entities
    base = str(site.base_url).rstrip("/")
    return {
        "name": site.name,
        "url": site.base_url,
        "knowledge": {
            "llms_txt": f"{base}/llms.txt",
            "llms_full": f"{base}/llms-full.txt",
            "graph": f"{base}/graph.json",
            "schema_organization": f"{base}/schema/organization.jsonld",
            "schema_services": f"{base}/schema/services.jsonld",
        },
    }
